download/oci_dowload.go

```diff
@@ -248,25 +250,38 @@ func (d *OCIDownloader) download(ctx context.Context, m metrics.Metrics) (*downl
 	if tarballDescriptor.MediaType == "" {
 		return nil, fmt.Errorf("no tarball descriptor found in the layers")
 	}
-
-	bundleFilePath := filepath.Join(d.localStorePath, "blobs", "sha256", string(tarballDescriptor.Digest.Hex()))
+	etag := string(tarballDescriptor.Digest.Hex())
+	bundleFilePath := filepath.Join(d.localStorePath, "blobs", "sha256", etag)
+	// if the downloader etag sha is the same with digest of the tarball it was already loaded
+	if d.etag == etag {
```

Comment:
To clarify if the image has not changed we won't end up re-downloading it again, correct ?

The benefit here seems to be we won't read/activate it but what happens during download would be helpful to understand ?

Reply:
The go oras library uses the docker resolver and makes a call to retrieve the manifest of an image based on its tag. That reply contains a SHA and if that SHA is not present in the local OCI store only then it’s downloaded.

Reply:
Thanks for clarifying @carabasdaniel ! If we can get some tests in to verify the end-to-end flow that would be helpful.